avm/utilities/pipelines/sharedScripts/Set-ModuleReadMe.ps1

```diff
@@ -225,245 +225,269 @@ Child-level invocation during recursion.
-                $bicepAllowedValues = ConvertTo-FormattedBicep -JSONParameters $rawAllowedValues
-                $allowedValues = "{`n$bicepAllowedValues`n}"
-            } else {
-                $allowedValues = $rawAllowedValues
-            }
-
-            if (-not [String]::IsNullOrEmpty($allowedValues)) {
-                if (($allowedValues -split '\n').count -eq 1) {
-                    $formattedAllowedValues = '- Default: `{0}`' -f $allowedValues
-                } else {
-                    $formattedAllowedValues = @(
-                        '- Allowed:',
-                        '  ```Bicep',
+            '  ```'
+          )
+        }
+      } else {
+        $formattedDefaultValue = $null
+      }
+
+      # Format allowed values
+      # =====================
+      if ($rawAllowedValues -is [array]) {
+        $bicepJSONAllowedParameterObject = @{ $parameter.name = ($rawAllowedValues ?? @()) } # Wrapping on object to work with formatted Bicep script
+        $bicepRawformattedAllowed = ConvertTo-FormattedBicep -JSONParameters $bicepJSONAllowedParameterObject
+        $leadingSpacesToTrim = ($bicepRawformattedAllowed -match '^(\s+).+') ? $matches[1].Length : 0
+        $bicepCleanedFormattedAllowed = $bicepRawformattedAllowed -replace ('{0}: ' -f $parameter.name) # Unwrapping the object
+        $allowedValues = $bicepCleanedFormattedAllowed -split '\n' | ForEach-Object { $_ -replace "^\s{$leadingSpacesToTrim}" }  # Removing excess leading spaces
+      } elseif ($rawAllowedValues -is [hashtable]) {
+        $bicepAllowedValues = ConvertTo-FormattedBicep -JSONParameters $rawAllowedValues
+        $allowedValues = "{`n$bicepAllowedValues`n}"
+      } else {
+        $allowedValues = $rawAllowedValues
+      }
+
+      if (-not [String]::IsNullOrEmpty($allowedValues)) {
+        if (($allowedValues -split '\n').count -eq 1) {
+          $formattedAllowedValues = '- Default: `{0}`' -f $allowedValues
+        } else {
+          $formattedAllowedValues = @(
+            '- Allowed:',
+            '  ```Bicep',
                 ($allowedValues -split '\n' | Where-Object { -not [String]::IsNullOrEmpty($_) } | ForEach-Object { "  $_" } | Out-String).TrimEnd(),
-                        '  ```'
-                    )
-                }
-            } else {
-                $formattedAllowedValues = $null
-            }
-
-            # Build list item
-            # ===============
-            $listSectionContent += @(
-                $paramHeader,
+            '  ```'
+          )
+        }
+      } else {
+        $formattedAllowedValues = $null
+      }
+
+      # Format example
+      # ==============
+      if (-not [String]::IsNullOrEmpty($example)) {
+        # allign content to the left by removing trailing whitespaces
+        $leadingSpacesToTrim = ($example -match '^(\s+).+') ? $matches[1].Length : 0
+        $bicepCleanedExample = $example -replace ('{0}: ' -f $parameter.name) # Unwrapping the object
```

Comment:
Why the cleanup? Is it for cases where somebody creates an example like
```bicep
@metadata({
  domainName : 'contoso.onmicrosoft.com'
})
param domainName string
```
where the parameter name is part of the example?

Reply:
Good catch. Actually this is removing too much from my examples. Removing...